### DISCUSSION TITLE
Optimization of loadUserSessionsWithClientSessions method

### DISCUSSION THREAD

**Discussion by ArmanGhost** [keycloak/keycloak#11779] [Ideas]
We have a problem with getting of all user offline sessions in out SPI. After code inspection we found that part of code
![image](https://user-images.githubusercontent.com/39274014/165990465-35af780c-5279-4229-8c24-24f99d3e6c12.png)
For example, if you need to get two user offline sessions you need to select all offline client sessions beetween them. And in our case it can be 100 000 of them. Our keycloak is crashed vecause of that.
After we disable getting offline user profiles, our problem is solved, but we need this
URL of this commit:
https://github.com/keycloak/keycloak/commit/919899b99420c6821f30f3d44ee091e632e88e20
@hmlnarik and @pfstrifork can you help us with this?
We changed a code to last version (before some changes that which led to this non optimization query)
![image](https://user-images.githubusercontent.com/39274014/165991008-fd7729dc-86ee-42e9-a8dc-7eb26ece8553.png)

**Comment by AugustoSansoni** [comment]
The current query filters within a range of UUIDs and can load an indefinite number of records. It seems very odd. Is there any good reason to do so ?

**Comment by hmlnarik** [comment]
This issue will be treated in issue #13340
